de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.swtbot.test;
+
+
+import java.util.List;
+
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefConnectionEditPart;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditPart;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditor;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.concept.unittest.util.ConceptXmiLoader;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.Interface;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceEnd;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceType;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceTypeCollection;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+import org.junit.Assert;
+
+/**
+ * 
+ * SWTBot test class for testing Funcelectrical Diagrams
+ *
+ */
+public class FuncElectricalDiagramTest extends ASwtBotTestCase {
+	
+	private SWTBotTreeItem repositoryNavigatorItem;
+	private SWTBotTreeItem configurationTree;
+	private SWTBotTreeItem elementConfiguration;
+	private SWTBotGefEditor diagramEditor;
+	
+	private Concept conceptFuncElectrical;
+	
+	private static final int DRAG1_COORDINATES_X = 100;
+	private static final int DRAG1_COORDINATES_Y = 100;
+	
+	private static final int DRAG2_COORDINATES_X = 100;
+	private static final int DRAG2_COORDINATES_Y = 200;
+
+	@Before
+	public void before() throws Exception {
+		super.before();
+		// create the necessary items for the test
+		conceptFuncElectrical = ConceptXmiLoader.loadConceptFromPlugin(de.dlr.sc.virsat.model.extension.funcelectrical.Activator.getPluginId() + "/concept/concept.xmi");
+		repositoryNavigatorItem = bot.tree().expandNode(SWTBOT_TEST_PROJECTNAME, "Repository");
+		configurationTree = addElement(ConfigurationTree.class, conceptPs, repositoryNavigatorItem);
+		elementConfiguration = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+				
+		createNewDiagramForTreeItem(configurationTree, DiagramType.interfaces);
+		diagramEditor = getOpenedDiagramEditorbyTitle("newDiagram");
+	}
+	
+	@Test
+	public void addInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
```

Comment:
The name ElementConfiguration is used quite often.
Maybe we can move it into a constant? If it were to change for some reason, then it would be easy to update.